storage/database/db_manager_test.go

```diff
@@ -672,24 +673,212 @@ func TestDBManager_Governance(t *testing.T) {
 	// TODO-Klaytn-Database Implement this!
 }
 
-func TestDBManager_StateTrieMigration(t *testing.T) {
+func TestDatabaseManager_CreateMigrationDBAndSetStatus(t *testing.T) {
+	for i, dbm := range dbManagers {
+		if dbConfigs[i].DBType == MemoryDB {
+			continue
+		}
+
+		// check if migration fails on single DB
+		if !dbm.IsPartitioned() {
+			migrationBlockNum := uint64(12345)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration())
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err) // expect error
+
+			continue
+		}
+
+		// check if migration fails when in migration
+		{
+			migrationBlockNum := uint64(34567)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration())
+
+			// set migration status
+			dbm.setStateTrieMigrationStatus(migrationBlockNum)
+			assert.True(t, dbManagers[i].InMigration())
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err) // expect error
+
+			// reset migration status for next test
+			dbm.setStateTrieMigrationStatus(0)
+		}
+
+		// check if CreateMigrationDBAndSetStatus works as expected
+		{
+			migrationBlockNum := uint64(56789)
+
+			// check if not in migration state
+			assert.False(t, dbManagers[i].InMigration())
+
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.NoError(t, err)
+
+			// check if in migration state
+			assert.True(t, dbm.InMigration())
+
+			// check migration DB path in MiscDB
+			migrationDBPathKey := append(databaseDirPrefix, common.Int64ToByteBigEndian(uint64(StateTrieMigrationDB))...)
+			fetchedMigrationPath, err := dbm.getDatabase(MiscDB).Get(migrationDBPathKey)
+			assert.NoError(t, err)
+			expectedMigrationPath := dbBaseDirs[StateTrieMigrationDB] + "_" + strconv.FormatUint(migrationBlockNum, 10)
+			assert.Equal(t, expectedMigrationPath, string(fetchedMigrationPath))
```

Comment:
We better use a predefined expected value since it is generating an expected value using almost same logic with tested function.

Reply:
I changed to have less logical code in expected codes. PTAL :)